### PR TITLE
Treat error in addMessages as non-fatal

### PR DESCRIPTION
**Pull request by joshuacolvin0**

### CODE REVIEW

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 1]
```diff
@@ -110,13 +110,28 @@ bool ArbCore::machineIdle() {
...
-    if (!core_error && current_status != MESSAGES_READY) {
+    if (current_status != MESSAGES_ERROR && current_status != MESSAGES_READY) {
         message_data_status = MESSAGES_EMPTY;
     }
+
     return current_status;
 }
 
+std::string ArbCore::messagesClearError() {
+    auto lock = std::shared_lock<std::shared_mutex>(message_data_error_mutex);
```
Comment (PlasmaPower) [inline packages/arb-avm-cpp/data_storage/src/arbcore.cpp:123]: Doesn't this need to be a unique lock as it mutates the `message_data_error_string`? If so, and we remove the lock in `messagesStatus`, we should probably make this a regular non-shared mutex.

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 2]
```diff
@@ -110,13 +110,28 @@ bool ArbCore::machineIdle() {
 }
 
 ArbCore::message_status_enum ArbCore::messagesStatus() {
+    auto lock = std::shared_lock<std::shared_mutex>(message_data_error_mutex);
```
[the remark this answers is not in the document]
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/src/arbcore.cpp]: as discussed, mutex can be removed

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp (hunk 1 above)
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/src/arbcore.cpp]: as discussed, mutex can be removed

File: packages/arb-avm-cpp/data_storage/include/data_storage/arbcore.hpp [hunk 3]
```diff
@@ -166,6 +166,11 @@ class ArbCore {
     // Core thread inbox status input/output. Core thread will update if and
     // only if set to MESSAGES_READY
     std::atomic<message_status_enum> message_data_status{MESSAGES_EMPTY};
+    std::string message_data_error_string;
+    // Message_data_error_mutex non-arbcore threads should acquire read mutex
+    // when checking for MESSAGES_ERROR, arbcore thread needs to acquire write
+    // mutex only when setting MESSAGES_ERROR
+    std::shared_mutex message_data_error_mutex;
```
Comment (PlasmaPower) [inline packages/arb-avm-cpp/data_storage/include/data_storage/arbcore.hpp:173]: I think this field can be removed now.

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 4]
```diff
@@ -111,12 +111,25 @@ bool ArbCore::machineIdle() {
...
 }
 
+std::string ArbCore::messagesClearError() {
+    if (message_data_status != ArbCore::MESSAGES_ERROR) {
+        return "";
+    }
+
+    message_data_status = MESSAGES_EMPTY;
+    auto str = message_data_error_string;
+    message_data_error_string.clear();
```
Comment (PlasmaPower) [inline packages/arb-avm-cpp/data_storage/src/arbcore.cpp:128]: We should clear `message_data_error_string` before setting `message_data_status` back to empty, to ensure that another error isn't set inbetween the status being set and the error being cleared.

**Review by PlasmaPower — CHANGES_REQUESTED** [review]

File: packages/arb-avm-cpp/data_storage/include/data_storage/arbcore.hpp (hunk 3 above)
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/include/data_storage/arbcore.hpp:173]: :+1:

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp (hunk 4 above)
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/src/arbcore.cpp:128]: :+1:

**Review by PlasmaPower — APPROVED** [review]
LGTM

**Pull request merged**